File: src/main/java/io/olkkani/question/service/QuestionService.java
package io.olkkani.question.service;

import io.olkkani.question.dto.QuestionRequest;
import io.olkkani.question.entity.Question;
import java.util.List;
import org.springframework.data.domain.Pageable;

public interface QuestionService {

    public void save (QuestionRequest request);
    public List<QuestionRequest> findAll ();

    public QuestionRequest findById (Long id);

    public List<QuestionRequest> findRandomQuestions ();

    public void deleteById (Long id);
//    public List<QuestionRequest> find(Pageable pageable);
}

File: src/main/kotlin/io/olkkani/cleancodeleveltest/service/QuestionService.kt
package io.olkkani.cleancodeleveltest.service

import io.olkkani.cleancodeleveltest.config.exception.NotFoundException
import io.olkkani.cleancodeleveltest.domain.Question
import io.olkkani.cleancodeleveltest.domain.QuestionRepository
import io.olkkani.cleancodeleveltest.model.toResponse
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class QuestionService (
    private val questionRepository: QuestionRepository
){
        @Transactional(readOnly = true)
    fun getAll() =
            questionRepository.findAll().map { it.toResponse() }
    fun get(id: Long): Question {
        return questionRepository.findByIdOrNull(id) ?: throw NotFoundException("질문이 존재하지 않습니다.")
    }
}